spark_log_parser/eventlog.py

```diff
@@ -23,51 +23,63 @@ def _validate_work_dir(self, work_dir: Path | str) -> Path:
 
-        dat = []
-        for log in event_logs:
-            with open(log) as log_file:
+    def _get_event_log(self, paths: list[Path]) -> Path:
+        log_files = []
+        rollover_dat = []
+        for path in paths:
+            with open(path) as fobj:
                 try:
-                    line = json.loads(log_file.readline())
+                    line = json.loads(fobj.readline())
                 except ValueError:
-                    continue  # Maybe a Databricks pricing file
-                if line["Event"] == "DBCEventLoggingListenerMetadata":
-                    dat.append((line["Rollover Number"], line["SparkContext Id"], log))
-                else:
-                    raise ValueError("Expected DBC event not found")
+                    continue
+                if "Event" in line:
+                    log_files.append(path)
+                    if line["Event"] == "DBCEventLoggingListenerMetadata":
+                        rollover_dat.append(
+                            (line["Rollover Number"], line["SparkContext Id"], path)
+                        )
+
+        if rollover_dat:
+            if len(log_files) > len(rollover_dat):
+                raise ValueError("No rollover properties found in log file")
+
+            return self._concat(rollover_dat)
+
+        if len(log_files) > 1:
+            raise ValueError("No rollover properties found in log file")
+
+        return log_files[0]
+
+    def _concat(self, rollover_dat: list[tuple[str, str, str]]) -> Path:
+        rollover_df = pd.DataFrame(
+            rollover_dat, columns=["rollover_index", "context_id", "path"]
+        ).sort_values("rollover_index")
+
+        if not len(rollover_df.context_id.unique()) == 1:
+            raise ValueError("Not all rollover log files have the same Spark context ID")
 
-        df = pd.DataFrame(dat, columns=["rollover_index", "context_id", "path"]).sort_values(
-            "rollover_index"
-        )
+        diffs = rollover_df.rollover_index.diff()
 
-        self._validate_rollover_logs(df)
+        if any(diffs > 1) or rollover_df.rollover_index[0] > 0:
+            raise ValueError("Rollover log file appears to be missing")
+
+        if any(diffs < 1):
+            raise ValueError("Duplicate rollover log file detected")
 
         event_log = Path(tempfile.mkstemp(suffix="-concatenated.json", dir=str(self.work_dir))[1])
         with open(event_log, "w") as fobj:
-            for path in df.path:
+            for path in rollover_df.path:
                 with open(path) as part_fobj:
                     for line in part_fobj:
                         fobj.write(line)
 
         return event_log
-
-    def _validate_rollover_logs(self, df: pd.DataFrame):
-        if not len(df.context_id.unique()) == 1:
-            raise ValueError("Not all rollover files have the same Spark context ID")
-
-        diffs = df.rollover_index.diff()[1:]
-
-        if any(diffs > 1) or df.rollover_index[0] > 0:
```

Comment:
I remember seeing this in the last PR to check for the case this PR is addressing. Did this logic just not get run if there was only one log?

Reply:
Exactly. With only 1 log file rollover validation was skipped.